service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/notebook/ServiceAccountPolicyStep.java

```diff
@@ -102,11 +105,15 @@ private Binding createWriterBinding(FlightMap workingMap) throws InterruptedExce
         privateIamRoles.stream().anyMatch(role -> role.equals(ControlledResourceIamRole.WRITER)),
         "A private, controlled AI Notebook instance must have the writer role or else it is not useful.");
 
-    // Assumes that the assigned user is the user requesting the resource creation.
-    String userEmail = samService.getRequestUserEmail(userRequest);
+    // Grant permission via private resource's Sam group so that access can be revoked via Sam.
+    Map<ControlledResourceIamRole, String> resourceRoleGroupsMap =
+        workingMap.get(
+            WorkspaceFlightMapKeys.ControlledResourceKeys.IAM_RESOURCE_GROUP_EMAIL_MAP,
+            new TypeReference<>() {});
+    String writerGroupEmail = resourceRoleGroupsMap.get(ControlledResourceIamRole.WRITER);
     return new Binding()
         .setRole(SERVICE_ACCOUNT_USER_ROLE)
-        .setMembers(ImmutableList.of("user:" + userEmail));
+        .setMembers(ImmutableList.of("group:" + writerGroupEmail));
```

Comment:
I think `Collections.singletonList()` might be preferred, but I don't recall why.

Reply:
SGTM for a one element list